src/app/reset-password/reset-password.component.ts

```diff
@@ -55,6 +56,14 @@ export class ResetPasswordComponent implements OnInit {
     this.route.queryParams
       .subscribe(params => {
         this.token = params.token;
+        this.identityService.checkTokenValid(this.token).subscribe(
```

Comment:
of course if the backend is slow to respond or times out, the user will first see the new password form and then see it disappear and be replaced with the error message. Not sure what the best way to avoid that is, or if we should just live with it.

Reply:
@bdsl this is a really good consideration. To work around this, we can simply use a `<mat-spinner>`, like we use elsewhere in the app. The spinner should show whilst the server does its processing, until it returns valid or invalid.
To do this, we can default `tokenValid` to undefined, like this:
`tokenValid: boolean|undefined = true;`
Then, we can render the `<mat-spinner>` in the template file while `tokenValid === undefined`

Reply:
Thanks, done in 2c4f0028

Reply:
Awesome!